x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/action/role/TransportQueryRoleAction.java

```diff
@@ -88,12 +88,14 @@ protected void doExecute(Task task, QueryRoleRequest request, ActionListener<Que
         if (accessesRoleName.get()) {
             searchSourceBuilder.runtimeMappings(ROLE_NAME_RUNTIME_MAPPING);
         }
-        nativeRolesStore.queryRoleDescriptors(
-            searchSourceBuilder,
-            ActionListener.wrap(
-                queryRoleResults -> listener.onResponse(new QueryRoleResponse(queryRoleResults.total(), queryRoleResults.items())),
-                listener::onFailure
-            )
-        );
+        nativeRolesStore.ensureBuiltinRolesAreQueriable(ActionListener.wrap(onResponse -> {
```

Comment:
The built-in reserved roles are maybe indexed just before the query roles action is executed.
Normally, the query-roles action is a "read" type of action, but this change makes it a possible write.